src/Adapter/Form/ChoiceProvider/TaxRuleGroupChoiceProvider.php

```diff
@@ -53,26 +75,32 @@ public function getChoices()
      */
     public function getChoicesAttributes(): array
     {
-        $attrs = [];
+        $address = new Address();
+        $address->id_country = $this->countryId;
+
+        $taxRates = [];
         foreach ($this->getRules() as $rule) {
-            // Keep first one found
-            if (!empty($attrs[$rule['name']]['data-tax-rate'])) {
-                continue;
-            }
+            $taxRulesGroupId = new TaxRulesGroupId((int) $rule['id_tax_rules_group']);
+            $stateId = $this->taxRulesGroupRepository->getTaxRulesGroupDefaultStateId($taxRulesGroupId, new CountryId($this->countryId));
+            $address->id_state = $stateId;
 
-            $attrs[$rule['name']] = [
-                'data-tax-rate' => !empty($rule['rate']) ? $rule['rate'] : 0,
+            $taxManagerFactory = TaxManagerFactory::getManager($address, $taxRulesGroupId->getValue());
+            $taxRate = $taxManagerFactory->getTaxCalculator()->getTotalRate();
+            $state = new State($taxManagerFactory->getAddress()->id_state);
```

Comment:
```suggestion
            $state = new State($stateId);
```

Reply:
```suggestion
            $state = new State($address->id_state);
```

Reply:
You have two other possibilities to get this data so no need to add a function on the tax manager interface which is not useful and out of the initial scope of the interface

Reply:
Besides you should only fetch the state when `$stateId` is different from 0 this will avoid many useless queries (as most countries don't have states).

```suggestion
            $stateIsoCode = '';
            if ($stateId > 0) {
                $state = new State($stateId);
                $stateIsoCode = $state->iso_code;
            }
```